libs/top_level_metrics.py

```diff
@@ -87,21 +98,40 @@ def calculate_test_positivity(
     """
     daily_negative_tests = negative_tests.diff()
     daily_positive_tests = positive_tests.diff()
-
     positive_smoothed = series_utils.smooth_with_rolling_average(daily_positive_tests)
     negative_smoothed = series_utils.smooth_with_rolling_average(
         daily_negative_tests, include_trailing_zeros=False
     )
 
     last_n_positive = positive_smoothed[-lag_lookback:]
     last_n_negative = negative_smoothed[-lag_lookback:]
-    # TODO: Porting from: https://github.com/covid-projections/covid-projections/blob/master/src/common/models/Projection.ts#L521.
-    # Do we still want to return no data if there appears to be positive case data but lagging data for negative cases?
+
     if any(last_n_positive) and last_n_negative.isna().all():
         return pd.Series([], dtype="float64")
     return positive_smoothed / (negative_smoothed + positive_smoothed)
 
 
+def calculate_contact_tracers(
+    cases: pd.Series,
+    contact_tracers: pd.Series,
+    contact_tracers_per_case: int = CONTACT_TRACERS_PER_CASE,
+) -> pd.Series:
+    """Calculates ratio of hired tracers to estimated tracers needed based on daily cases.
+
+    Args:
+        cases: Cumulative cases.
```

Comment:
Do these Series have datetime index? no index but already aligned?

Reply:
unfortunately they don't have a datetime index.  I went back and forth on wether to make it so or not.  the ultimate reason I held off was because it was a bit more work to make the functions in `series_util` (for example smoothing) to work the same on data w/ a datetime index and data without. but definitely up for changing it in the future.

Reply:
but correct, they are already aligned

Reply:
cool. my comment was more a request to document the existing state then change the structure in this PR.